provision/kubernetes/provisioner.go

```diff
@@ -931,63 +960,93 @@ func (p *kubernetesProvisioner) DeleteVolume(volumeName, pool string) error {
 	if err != nil {
 		return err
 	}
-	return deleteVolume(client, volumeName, client.Namespace(pool))
+	return deleteVolume(client, volumeName, client.PoolNamespace(pool))
 }
 
 func (p *kubernetesProvisioner) IsVolumeProvisioned(volumeName, pool string) (bool, error) {
 	client, err := clusterForPool(pool)
 	if err != nil {
 		return false, err
 	}
-	return volumeExists(client, volumeName, client.Namespace(pool))
+	return volumeExists(client, volumeName, client.PoolNamespace(pool))
+}
+
+func (p *kubernetesProvisioner) UpdateApp(old, new provision.App, w io.Writer) error {
+	if old.GetPool() == new.GetPool() {
+		return nil
+	}
+	client, err := clusterForPool(old.GetPool())
+	if err != nil {
+		return err
+	}
+	newclient, err := clusterForPool(new.GetPool())
+	if err != nil {
+		return err
+	}
+	sameCluster := client.GetCluster().Name == newclient.GetCluster().Name
+	params := updatePipelineParams{
+		old: old,
+		new: new,
+		w:   w,
+		p:   p,
+	}
+	if !sameCluster {
```

Comment:
[nit] why don't you use `if client.GetCluster().Name == newclient.GetCluster().Name {` directly? This `sameCluster` variable isn't used elsewhere